node/subsystem-util/src/runtime/mod.rs

```diff
@@ -100,14 +102,78 @@ impl Default for Config {
 	}
 }
 
-impl RuntimeInfo {
+/// Runtime information provider
+#[async_trait]
+pub trait RuntimeInfoProvider {
```

Comment:
Why do we need this? For testing, we can also just intercept the messages it is sending as we usually do.

Reply:
Can you please link an example?

Having a mock for tests seems to be more idiomatic anyway, doesn't it?

Reply:
Just normal mocking of messages: https://github.com/paritytech/polkadot/blob/9b032392a0c6983067773d10ebce493200a73b3b/node/network/dispute-distribution/src/tests/mod.rs#L791

Reply:
Yes, you're right! I didn't understand how it works and, as a result, overengineered it a bit :sweat_smile: 

Now it looks much less messy. Thank you!